jenkins/JenkinsfileUpdateEB

```diff
@@ -0,0 +1,375 @@
+                            status=0
+                            new_recipe=\$(echo \$recipe | sed -E 's/(Cray[a-zA-Z]+)-${params.cdt_current}/\\1-${params.cdt_version}/')
+                            echo -e "\nCurrent Update: \$new_recipe"
+
+                            # get relative path of dependencies from dry run
+                            deps=\$($build_command -D \$recipe | awk '/\\.eb/{printf"%s%s\\n","easyconfigs",gensub(/.*(\\/\\w\\/.*\\.eb).*/,"\\\\1",1)}' | sed -E 's/(Cray[a-zA-Z]+)-${params.cdt_current}/\\1-${params.cdt_version}/')
+                            
+                            # check status of dry run
+                            if [ -n "\$deps" ]; then
+                                $build_command $tc_flags -D \$recipe | grep "\\["
+                            else
+                                failed+=(\$new_recipe)
+                                echo "\$new_recipe" >> "\${failed_file}"
+                                echo -e "\nDry run of \$new_recipe FAILED: skipping build"
+                                continue
+                            fi
+
+                            echo -e "\nSubmitting job and building..."
+                            if [[ \$recipe =~ Cray[a-zA-Z]+-${params.cdt_current} ]]; then
+                                $build_command $tc_flags \$recipe
+                            else
+                                $build_command \$recipe
+                            fi
+                            status=\$[status+\$?]
+
+                            # keep track of successful and failed builds
+                            if [ \$status -eq 0 ]; then
+                                success+=(\$new_recipe)
+                                echo "\$new_recipe" >> "\${success_file}"
+                                echo "\$deps" | tr ' ' '\n' >> "\${submit_file}"
+                                echo -e "\nUpdate of \$new_recipe from \$recipe SUCCESSFUL(\${#success[@]})"
+                            else
+                                failed+=(\$new_recipe)
+                                echo "\$folder/\$new_recipe" >> "\${failed_file}"
+                                echo -e "\nUpdate of \$new_recipe from \$recipe FAILED(\${#failed[@]})"
+                            fi
+                        done
+
+                        # change permission of tmp folders if failed is not null
+                        if [ -n "\$failed" ]; then
+                            chmod -R o+r "\${EASYBUILD_TMPDIR}"
+                            find "\${EASYBUILD_TMPDIR}" -type d -exec chmod o+x '{}' \\;
+                        fi
+
+                        # print successful and failed updates
+                        echo -e "\nSuccessful Updates: \${#success[@]} \n\${success[@]}"
+                        echo -e "\nFailed Updates: \${#failed[@]} \n\${failed[@]}"
+
+                        # return the number of failed updates
+                        exit \${#failed[@]}
```

Comment:
Is Jenkins going to continue to the next steps if you return != 0 ?

Reply:
Yes, the next stages `Jira` and `GitHub` should be executed anyway